Missionframework/modules/05_garrison/fnc/fn_garrison_getVehSpawnPos.sqf

```diff
@@ -0,0 +1,41 @@
+/*
+    KPLIB_fnc_garrison_getVehSpawnPos
```

Comment:
I've thought about moving this to a `common` function.
My concern was that it has the "hardcoded" distances and randomization which are needed for the garrison but you shouldn't want if you like to get an emptyPosition as closest to a given position.
Moving it to `common` would need to make the range/randomization stuff variable with parameters. I thought this could be "too much" for such a function, but it could be done with sensible default values. 🤔 
Further Opinion?

Reply:
Ok, lets leave this as is for now. I'll take a look at it next sprint, three while loops in single function make me think that the whole function could be optimized further.

Reply:
Yeah, it's not looking very nice. But it's kind of needed from my current perspective, as `findEmptyPosition` returns `[]` when no position is found.
So first loop to get an initial position.
Next loop checks if the surface is water which takes only 2d or 3d position array. If it's water it uses findEmptyPosition again which could return `[]` and this would break the check entirely. That's the reason for the additional `isEqualTo []` loop.
If you (or anyone else who's reading this, of course) stumble upon a nicer and working solution, I really appreciate any input.

Reply:
Maybe consider replacing:
```sqf
while {surfaceIsWater _spawnPos} do {...
```
with:
```sqf
if (surfaceIsWater _spawnPos) exitWith {_this call KPLIB_fnc_garrison_getVehSpawnPos};
```

Reply:
`surfaceIsWater _spawnPos || _spawnPos isEqualTo []`
if it's equal to [] the surfaceIsWater would directly break the function due to invalid input arguments. I've tried it locally before commiting and it was the reason for the `[]` check inside the water-check.

Reply:
This should be fixable with lazy evaluation (code is not evaluated if first element returns true)
```sqf
_spawnPos isEqualTo [] || {surfaceIsWater _spawnPos};
```

Reply:
Lazy evaluation should be fine, yeah.
This would eliminate all loops and create a small and nice recursive function.
Guess this even not need to be tested concerning performance to the current implementation. 👍

Reply:
If executed in middle of the sea and executed unscheduled, as it is not using `while` loops which have iteration limit (in unscheduled) but recursion, it can freeze the game for looooong time.

Reply:
Maybe a limiter?
a _counter argument which is default 1.
And in the recursive we do the next call always with a
```
[_center, (_counter + 1)] call KPLIB_fnc_garrison_getVehSpawnPos;
```
Also adding a `if (_counter > 20) exitWith {[]};` to limit the execution.
Guess we've to test it and of course we need to provide a detection of boat vehicles or a seperate "spawnBoat" function.